pegen/grammar.py

```diff
@@ -386,192 +386,22 @@ def initial_names(self) -> AbstractSet[str]:
-            icut = len(items)
-            mark = self.mark()
-            while item := self.named_item():
-                assert item
-                items.append(item)
-                mark = self.mark()
-        if self.expect('$'):
-            items.append(NamedItem(None, NameLeaf('ENDMARKER')))
-        if icut == len(items):
-            # Can't have "cut" as the last item
-            self.reset(ubermark)
-            return None
-        action = self.curly_stuff()
-        return Alt(items, icut=icut, action=action.string if action else None)
-
-    @memoize
-    def named_item(self) -> Optional[NamedItem]:
-        """
-        named_item: NAME '=' item | item | lookahead
-        """
-        mark = self.mark()
-        if (name := self.name()) and self.expect('=') and (item := self.item()):
-            assert name
-            assert item
-            return NamedItem(name.string, item)
-        self.reset(mark)
-        item = self.item()
-        if not item:
-            self.reset(mark)  # Redundant?
-            item = self.lookahead()
-            if not item:
-                return None
-        return NamedItem(None, item)
-
-    @memoize
-    def lookahead(self) -> Optional[Lookahead]:
-        """
-        lookahead: ('&' | '!') atom
-        """
-        mark = self.mark()
-        if (lookahead := (self.expect('&') or self.expect('!'))) and (atom := self.atom()):
-            assert lookahead
-            assert atom
-            if lookahead.string == '&':
-                return PositiveLookahead(atom)
-            else:
-                return NegativeLookahead(atom)
-        self.reset(mark)
-        return None
-
-    @memoize
-    def item(self) -> Optional[Item]:
-        """
-        item: '[' alternatives ']' | atom ('?' | '*' | '+')?
-        """
-        mark = self.mark()
-        if self.expect('[') and (alts := self.alternatives()) and self.expect(']'):
-            assert alts
-            return Opt(alts)
-        self.reset(mark)
-        if atom := self.atom():
-            assert atom
-            mark = self.mark()
-            if self.expect('?'):
-                return Opt(atom)
-            if self.expect('*'):
-                return Repeat0(atom)
-            if self.expect('+'):
-                return Repeat1(atom)
-            return atom
-        return None
-
-    @memoize
-    def atom(self) -> Optional[Plain]:
-        """
-        atom: '(' alternatives ')' | NAME | STRING
-        """
-        mark = self.mark()
-        if self.expect('(') and (alts := self.alternatives()) and self.expect(')'):
-            assert alts
-            return Group(alts)
-        self.reset(mark)
-        if name := self.name():
-            assert name
-            return NameLeaf(name.string)
-        if string := self.string():
-            assert string
-            return StringLeaf(string.string)
-        return None
+Item = Union[Plain, Opt, Repeat, Lookahead, Rhs]
```

Comment:
Can you configure your editor to always ensure a file ends in \n (unless empty)?